File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/DocumentSeries/domain/DocumentSeries.vb

<Serializable()> _
Public Class DocumentSeries
    Inherits DomainObject(Of Int32)
    Implements IAuditable

    Public Overridable Property Name As String
    Public Overridable Property Container() As Container
    Public Overridable Property Family() As DocumentSeriesFamily
    Public Overridable Property PublicationEnabled() As Boolean?
    Public Overridable Property SubsectionEnabled() As Boolean?
    Public Overridable Property RoleEnabled() As Boolean?
    Public Overridable Property AllowNoDocument() As Boolean?
    Public Overridable Property AllowAddDocument() As Boolean?
    Public Overridable Property AttributeSorting() As String
    Public Overridable Property AttributeCache() As String
    Public Overridable Property RegistrationUser() As String Implements IAuditable.RegistrationUser
    Public Overridable Property RegistrationDate() As DateTimeOffset Implements IAuditable.RegistrationDate
    Public Overridable Property LastChangedUser As String Implements IAuditable.LastChangedUser
    Public Overridable Property LastChangedDate As DateTimeOffset? Implements IAuditable.LastChangedDate
    Public Overridable Property SortOrder() As Integer?
End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/DocumentArchives/DocumentSeries.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Entity.Commons;
using VecompSoftware.DocSuiteWeb.Entity.Resolutions;

namespace VecompSoftware.DocSuiteWeb.Entity.DocumentArchives
{
    public class DocumentSeries : DSWBaseEntity
    {
        #region [ Constructor ]

        public DocumentSeries() : this(Guid.NewGuid()) { }
        public DocumentSeries(Guid uniqueId)
            : base(uniqueId)
        {
            DocumentSeriesItems = new HashSet<DocumentSeriesItem>();
            DocumentSeriesConstraints = new HashSet<DocumentSeriesConstraint>();
            ResolutionKindDocumentSeries = new HashSet<ResolutionKindDocumentSeries>();
        }
        #endregion

        #region[ Properties ]

        public string Name { get; set; }
        public bool PublicationEnabled { get; set; }
        public bool? SubsectionEnabled { get; set; }
        public bool? RoleEnabled { get; set; }
        public bool? AllowNoDocument { get; set; }
        public bool? AllowAddDocument { get; set; }
        public bool? AttributeSorting { get; set; }
        public bool? AttributeCache { get; set; }
        public int? SortOrder { get; set; }

        //da togliere quando verrà mappata l'entità DocumentSeriesFamily
        public int? IdDocumentSeriesFamily { get; set; }

        #endregion

        #region[ Navigation Properties ]
        public virtual Container Container { get; set; }
        public virtual ICollection<DocumentSeriesItem> DocumentSeriesItems { get; set; }
        public virtual ICollection<DocumentSeriesConstraint> DocumentSeriesConstraints { get; set; }
        public virtual ICollection<ResolutionKindDocumentSeries> ResolutionKindDocumentSeries { get; set; }
        #endregion
    }
}
